mixins/interactive/interactive-mixin.js

```diff
@@ -98,25 +102,29 @@ export const InteractiveMixin = superclass => class extends LocalizeCoreElement(
 
 	_handleInteractiveContentFocusIn() {
 		this._interactive = true;
+		this._dismissibleId = setDismissible(async() => {
+			await this._handleInteractiveExit();
+		});
 	}
 
 	_handleInteractiveContentFocusOut(e) {
 		if (isComposedAncestor(this.shadowRoot.querySelector('.interactive-container-content'), e.relatedTarget)) return;
 		// focus moved out of the interactive content
 		this._interactive = false;
+		if (this._dismissibleId) {
+			clearDismissible(this._dismissibleId);
+			this._dismissibleId = null;
+		}
+	}
+
+	async _handleInteractiveExit() {
+		this._interactive = false;
+		await this.updateComplete;
+		this.shadowRoot.querySelector('.interactive-toggle').focus();
 	}
 
 	async _handleInteractiveKeyDown(e) {
-		if (!this._interactive) return;
-
-		if (e.keyCode === keyCodes.ESCAPE && !e.target?.opened) {
-			e.stopPropagation();
-			this._interactive = false;
-			await this.updateComplete;
-			this.shadowRoot.querySelector('.interactive-toggle').focus();
-		} else if (e.keyCode === keyCodes.TAB) {
-			e.stopPropagation();
-		}
+		if (this._interactive && e.keyCode === 9) e.stopPropagation();
```

Comment:
Nit: maybe we should add a comment here to indicate which key this is... `// TAB`